packages/main/src/ComboBox.ts

```diff
@@ -0,0 +1,1240 @@
+import "@ui5/webcomponents-icons/dist/not-editable.js";
+import "@ui5/webcomponents-icons/dist/error.js";
+import "@ui5/webcomponents-icons/dist/alert.js";
+import "@ui5/webcomponents-icons/dist/sys-enter-2.js";
+import "@ui5/webcomponents-icons/dist/information.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { getFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";
+import {
+	isBackSpace,
+	isDelete,
+	isShow,
+	isUp,
+	isDown,
+	isEnter,
+	isEscape,
+	isTabNext,
+	isTabPrevious,
+	isPageUp,
+	isPageDown,
+	isHome,
+	isEnd,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import * as Filters from "./Filters.js";
+
+import {
+	VALUE_STATE_SUCCESS,
+	VALUE_STATE_ERROR,
+	VALUE_STATE_WARNING,
+	VALUE_STATE_INFORMATION,
+	VALUE_STATE_TYPE_SUCCESS,
+	VALUE_STATE_TYPE_INFORMATION,
+	VALUE_STATE_TYPE_ERROR,
+	VALUE_STATE_TYPE_WARNING,
+	INPUT_SUGGESTIONS_TITLE,
+	SELECT_OPTIONS,
+	LIST_ITEM_POSITION,
+	LIST_ITEM_GROUP_HEADER,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Templates
+import ComboBoxTemplate from "./generated/templates/ComboBoxTemplate.lit.js";
+import ComboBoxPopoverTemplate from "./generated/templates/ComboBoxPopoverTemplate.lit.js";
+
+// Styles
+import ComboBoxCss from "./generated/themes/ComboBox.css.js";
+import ComboBoxPopoverCss from "./generated/themes/ComboBoxPopover.css.js";
+import ResponsivePopoverCommonCss from "./generated/themes/ResponsivePopoverCommon.css.js";
+import ValueStateMessageCss from "./generated/themes/ValueStateMessage.css.js";
+import SuggestionsCss from "./generated/themes/Suggestions.css.js";
+
+import ComboBoxItem from "./ComboBoxItem.js";
+import Icon from "./Icon.js";
+import Popover from "./Popover.js";
+import ResponsivePopover from "./ResponsivePopover.js";
+import List from "./List.js";
+import type { ClickEventDetail } from "./List.js";
+import BusyIndicator from "./BusyIndicator.js";
+import Button from "./Button.js";
+import StandardListItem from "./StandardListItem.js";
+import ComboBoxGroupItem from "./ComboBoxGroupItem.js";
+import GroupHeaderListItem from "./GroupHeaderListItem.js";
+import ComboBoxFilter from "./types/ComboBoxFilter.js";
+import type FormSupportT from "./features/InputElementsFormSupport.js";
+import PopoverHorizontalAlign from "./types/PopoverHorizontalAlign.js";
+
+const SKIP_ITEMS_SIZE = 10;
+
+type ValueStateAnnouncement = Record<ValueState, string | undefined>;
+type ValueStateTypeAnnouncement = Record<ValueState, string>;
+
+type ComboBoxListItem = StandardListItem & {
+	mappedItem: ComboBoxItem
+};
+
+enum ValueStateIconMapping {
+	Error = "error",
+	Warning = "alert",
+	Success = "sys-enter-2",
+	Information = "information",
+}
+
+type ComboBoxSelectionChangeEventDetail = {
+	item: ComboBoxItem,
+};
+
+interface IComboBoxItem extends UI5Element {
+	text: string,
+	focused: boolean,
+	selected: boolean
```

Comment:
this one could be option parameter in order to not implement it inside comboboxgroupitem